### PR TITLE
feat: Add `Tags` field to V1 Event model and V2 Event DTO and model

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

Issue Number:  #270

## What is the new behavior?
`Tags` field is on V1 Event model and V2 Event DTO and model

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?
no
## Other information

### CODE REVIEW

File: models/event.go [hunk 1]
```diff
@@ -22,14 +22,15 @@ import (
...
+	Device      string            `json:"device,omitempty" codec:"device,omitempty"`     // Device identifies the source of the event, can be a device name or id. Usually the device name.
+	Created     int64             `json:"created,omitempty" codec:"created,omitempty"`   // Created is a timestamp indicating when the event was created.
+	Modified    int64             `json:"modified,omitempty" codec:"modified,omitempty"` // Modified is a timestamp indicating when the event was last modified.
+	Origin      int64             `json:"origin,omitempty" codec:"origin,omitempty"`     // Origin is a timestamp that can communicate the time of the original reading, prior to event creation
+	Readings    []Reading         `json:"readings,omitempty" codec:"readings,omitempty"` // Readings will contain zero to many entries for the associated readings of a given event.
+	Tags        map[string]string `json:"tags,omitempty" codec:"tags,omitempty"`         // Tags is an optional collection of key/value pairs that all the event to be tagged with custom information.
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply models/event.go:32]: Good catch. Fixed.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

File: v2/dtos/event_test.go [hunk 2]
```diff
@@ -22,6 +22,9 @@ func TestFromEventModelToDTO(t *testing.T) {
 		DeviceName: TestDeviceName,
 		Created:    TestTimestamp,
 		Origin:     TestTimestamp,
+		Tags: map[string]string{
+			"GatewayID": "Intel123",
```
Comment (cloudxxx8) [inline v2/dtos/event_test.go:26]: The test data doesn't match the expected data
`"GatewayID": "Houston-0001"`

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v2/dtos/event_test.go (hunk 2 above)
Comment (lenny-intel) [reply v2/dtos/event_test.go]: Humm, thought I changed all of them to `Houston-0001`... :-(

Fixed and all tests passing in my branch. :-)

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**